v1/src/main/java/com/google/cloud/teleport/templates/SpannerVectorEmbeddingExport.java

```diff
@@ -0,0 +1,314 @@
+
+    @TemplateParameter.Text(
+        order = 20,
+        regexes = {".+"},
+        description = "Read data from Cloud Spanner Instance",
+        helpText = "Instance of requested table.")
+    ValueProvider<String> getSpannerInstanceId();
+
+    @SuppressWarnings("unused")
+    void setSpannerInstanceId(ValueProvider<String> spannerInstanceId);
+
+    @TemplateParameter.Text(
+        order = 30,
+        regexes = {".+"},
+        description = "Read data from Cloud Spanner Database ",
+        helpText = "Database of requested table.")
+    ValueProvider<String> getSpannerDatabaseId();
+
+    @SuppressWarnings("unused")
+    void setSpannerDatabaseId(ValueProvider<String> spannerDatabaseId);
+
+    @TemplateParameter.Text(
+        order = 40,
+        regexes = {"^.+$"},
+        description = "Spanner Table",
+        helpText = "Spanner Table to read from")
+    ValueProvider<String> getSpannerTable();
+
+    void setSpannerTable(ValueProvider<String> table);
+
+    @TemplateParameter.Text(
+        order = 50,
+        description = "Columns to Export from Spanner Table",
+        helpText =
+            "Comma separated list of columns which are required for Vertex AI Vector Search Index."
+                + " If the column names don't precisely align with the Vertex AI Vector Search Index input structure, you can establish column mappings using aliases."
+                + " For e.g. if you have columns id and my_embedding i.e. the id column matches what vertex expects but the embedding column is named differently,"
+                + " you can specify the following `id, my_embedding:embedding`.")
+    ValueProvider<String> getSpannerColumnsToExport();
+
+    void setSpannerColumnsToExport(ValueProvider<String> value);
+
+    @TemplateParameter.GcsWriteFolder(
+        order = 60,
+        description = "Output files folder in Cloud Storage",
+        helpText = "The Cloud Storage folder for writing output files. Must end with a slash.",
+        example = "gs://your-bucket/folder1/")
+    ValueProvider<String> getGcsOutputFolder();
+
+    void setGcsOutputFolder(ValueProvider<String> value);
+
+    @TemplateParameter.Text(
+        order = 70,
+        description = "Output files prefix in Cloud Storage",
+        helpText = "The filename prefix for writing output files.",
+        example = "vector-embeddings")
+    ValueProvider<String> getGcsOutputFilePrefix();
+
+    void setGcsOutputFilePrefix(ValueProvider<String> textWritePrefix);
+
+    @TemplateParameter.Text(
+        order = 80,
+        optional = true,
+        description = "Cloud Spanner Endpoint to call",
+        helpText =
+            "The Cloud Spanner endpoint to call in the template. The default is set to https://batch-spanner.googleapis.com.",
+        example = "https://batch-spanner.googleapis.com")
+    @Default.String("https://batch-spanner.googleapis.com")
+    ValueProvider<String> getSpannerHost();
```

Comment:
very nit: I feel those parameters exist in multiple places / templates. We could create something like CommonSpannerOptions to include things like endpoint, RPC priority, etc so it's easier to track

Reply:
Thanks @bvolpato  for the review. Earlier we have were extending out options from the existing [SpannerReadOptions](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java#L77) but we actually wanted to  change few things like: 
1. Snapshot to Version time 
2. Order of Fields etc. 

So we added the options in our custom class. I think if someone wants to use it from common, they can use existing [SpannerReadOptions](https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v1/src/main/java/com/google/cloud/teleport/templates/common/SpannerConverters.java#L77). Let me know what's your thoughts on it.